### PR TITLE
Cassandra 17171 trunk

### PR DESCRIPTION
**Pull request by bereng**

### CODE REVIEW

**Comment by bereng** [comment]
CI [j8](https://app.circleci.com/pipelines/github/bereng/cassandra/517/workflows/d8fb9650-e805-42a6-81f1-e9bb495d5493) 1K runs

File: test/unit/org/apache/cassandra/db/compaction/CompactionsBytemanTest.java [hunk 1]
```diff
@@ -127,9 +128,10 @@ public void testCompactingCFCounting() throws Throwable
         cfs.enableAutoCompaction();
 
         execute("INSERT INTO %s (k, c, v) VALUES (?, ?, ?)", 0, 1, 1);
-        assertEquals(0, CompactionManager.instance.compactingCF.count(cfs));
+        Util.spinAssertEquals(true, () -> {return CompactionManager.instance.compactingCF.count(cfs) == 0;}, 1);
```
Comment (bereng) [inline test/unit/org/apache/cassandra/db/compaction/CompactionsBytemanTest.java]: Given in 17169 we've had timeouts on 1s tasks maybe I should do 10s here. Serves the same purpose and will avoid false flaky positives.

Comment (adelapena) [reply test/unit/org/apache/cassandra/db/compaction/CompactionsBytemanTest.java]: Nit: can we simplify the lambda?
```suggestion
        Util.spinAssertEquals(true, () -> CompactionManager.instance.compactingCF.count(cfs) == 0, 1);
```

File: test/unit/org/apache/cassandra/db/compaction/CompactionsBytemanTest.java [hunk 2]
```diff
@@ -127,9 +128,10 @@ public void testCompactingCFCounting() throws Throwable
...
         cfs.forceBlockingFlush();
 
+        Util.spinAssertEquals(true, () -> {return CompactionManager.instance.compactingCF.count(cfs) == 0;}, 1);
```
Comment (adelapena) [inline test/unit/org/apache/cassandra/db/compaction/CompactionsBytemanTest.java]:
```suggestion
        Util.spinAssertEquals(true, () -> CompactionManager.instance.compactingCF.count(cfs) == 0, 1);
```

File: test/unit/org/apache/cassandra/db/compaction/CompactionsBytemanTest.java (hunk 1 above)
Comment (adelapena) [reply test/unit/org/apache/cassandra/db/compaction/CompactionsBytemanTest.java]: Increasing to 10s or even more sounds good given our experience with timeouts. I understand that the 1s timeout matches the 1s delay injected by the Byteman rule, is this right? In that case we should also increase the Byteman delay, so the delay >= timeout.

**Comment by adelapena** [comment]
[Here](https://app.circleci.com/pipelines/github/adelapena/cassandra/1193/workflows/03a665e4-8156-429c-82c2-91b11018aa7d/jobs/10721) are 5k additional repetitions of the patched test.

**Comment by bereng** [comment]
Woo 5K all green sgtm :-). Wdyt?

**Comment by adelapena** [comment]
Looks good to me too, and +1 to the last changes. I have started new repeated runs for [j8](https://app.circleci.com/pipelines/github/adelapena/cassandra/1199/workflows/5272f762-e1b0-48e4-91ee-45de22bd294b) and [j11](https://app.circleci.com/pipelines/github/adelapena/cassandra/1200/workflows/3973ae47-2700-495c-b225-4715992eb2fd) with the new increased delay and timeouts, if they succeed I think we'll be ready to commit.

**Comment by adelapena** [comment]
All green in the last runs :)

**Review by adelapena — APPROVED** [review]

**Comment by bereng** [comment]
Squashed for commit

**Pull request closed without merging**